src/objc/classes.rs

```diff
@@ -675,3 +676,10 @@ impl ObjC {
         }
     }
 }
+
+pub fn class_getName(env: &mut Environment, class: id) -> ConstPtr<u8> {
+    let class_ptr: ConstPtr<class_t> = ConstPtr::from_bits(class.to_bits());
+    let class_data: class_t = env.mem.read(class_ptr);
+    let class_name: ConstPtr<u8> = env.mem.read(class_data.data).name;
+    class_name
+}
```

Comment:
Unfortunately, this won't always work. For classes defined by touchHLE itself (i.e. all the classes in our framework implementations), the class allocation isn't `class_t` and its name only exists as a Rust string inside its host object. That could be changed, I suppose, but I think I'd prefer allocating such strings on-demand. I'm assuming that the exact shape of `class_t` is a detail app code isn't expected to rely on.

Reply:
With that in mind, by the way, `NSStringFromClass` could potentially be written so that it creates the `NSString` from the Rust string directly. I don't have an opinion on whether it should, though I will say it's more efficient that way.

Reply:
Didn't realize this at the time ._. . Should hopefully be fine now, it seems to have worked with my quick test at least.